airflow/executors/celery_executor.py

```diff
@@ -377,6 +385,49 @@ def _check_for_stalled_adopted_tasks(self):
             for key in timedout_keys:
                 self.change_state(key, State.FAILED)
 
+    @provide_session
+    def _clear_stuck_queued_tasks(self, session: Session = NEW_SESSION) -> None:
+        """
+        Tasks can get stuck in queued state in DB while still not in
+        worker. This happens when the worker is autoscaled down and
+        the task is queued but has not been picked up by any worker prior to the scaling.
+
+        In such situation, we update the task instance state to scheduled so that
+        it can be queued again. We chose to use task_adoption_timeout to decide
+        """
+        if not isinstance(app.backend, DatabaseBackend):
+            # We only want to do this for database backends where
```

Comment:
Interesting - we have seen this same issue when using a Redis results backend - is there a good explanation for why this would only happen with DB backends?

Reply:
I can't figure how it could be solved in Redis. Do you have a reproductive step for Redis just like in #19699

Reply:
Oh yes, atleast in that case, the comment should be clear with a proper TODO or just solved in this PR too for all the other backends.